### ISSUE TITLE
Support for Python 3.13

### ISSUE THREAD

**Issue by jrobinAV** [Avaiga/taipy#2167]
### Description

Taipy should support Python 3.13.

### Code of Conduct

- [x] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by jrobinAV** [comment]
Sorry @tri0n0, but this issue is not open to an external contribution as it requires some privileges to update some private assets and run some tests.

Thank you for your understanding.

**Comment by FlorianJacta** [comment]
A user asked for this support

**Comment by joaoandre-avaiga** [comment]
As of now we are blocked on this. We have two major dependencies that we are supporting older versions for compatibility issues:
 - Pyarrow(16.0): Databricks and Parquet DataNode, it would have to be at least on version 18.0
 - Pandas(1.3.5): Dataiku had some issues with Pandas 2, so we maintain the support for 1.3.5. The lowest pandas version that supports 3.13 is Pandas 2.2.3

We could think of alternatives for the testing and building process to try support this based on a conditional dependency, if theres enough priority, i.e., if more users require it.

**Comment by awhitford** [comment]
* [Pyarrow](https://pypi.org/project/pyarrow/) 19.0.1 is available ✅ 
* [Pandas](https://pypi.org/project/pandas/) 1.3.5? Do you mean [1.5.3](https://pypi.org/project/pandas/1.5.3/#files)?  Is it realistic that that old artifact will be available for Python 3.13? 🤔

**Comment by FlorianJacta** [comment]
If we are blocked on this issue, we should change the doc to say we are not compatible with 3.13 @joaoandre-avaiga @jrobinAV

**Comment by jrobinAV** [comment]
> * [Pandas](https://pypi.org/project/pandas/) 1.3.5? Do you mean [1.5.3](https://pypi.org/project/pandas/1.5.3/#files)?  Is it realistic that that old artifact will be available for Python 3.13? 🤔

Hi @awhitford ,

Thank you for your message.

I agree; it's not realistic. Actually, we wanted to give some time to Dataiku to support pandas 2.X. to get rid of old pandas versions while maintaining Dataiku compatibility. However, we may need to move on now, as it's been a while. 
Same for databricks.

We are going to discuss this issue again with the maintainer's team.

Thanks again,

**Comment by jrobinAV** [comment]
> If we are blocked on this issue, we should change the doc to say we are not compatible with 3.13 [@joaoandre-avaiga](https://github.com/joaoandre-avaiga) [@jrobinAV](https://github.com/jrobinAV)

You are right. Here are two PRs to review:
https://github.com/Avaiga/taipy-doc/pull/1293
https://github.com/Avaiga/taipy-doc/pull/1294

**Comment by AlexandreSajus** [comment]
Moving this to critical, new python users are installing Python 3.13 and trying Taipy out and it fails at import. Described in #2492

**Comment by czacharym** [comment]
Has there been any discussion with Databricks and Parquet DataNode to move on the 2.2.3+ version of pandas?  If there is no push, these teams will never move on.  Other projects that are seeking to use Taipy with 3.13 should not be held up because of Databricks.  It would cause those seeking to use Taipy with 3.13+ to look for alternatives to Taipy.

**Comment by VFLins** [comment]
Python 3.14 has been out for a while now, and there is still no support for 3.13. For me, this makes Taipy look like legacy software and causes it to be disregarded as an alternative for new projects, or make even current ones migrate to other backends.

**Comment by czacharym** [comment]
I've basically pushed Taipy to the side.  I have no use for it, since they are not keeping pace with the more recent major releases (3.13, 3.14).  If a project requires one of the two releases of Python, Taipy would be considered obsolete.

**Comment by arcanaxion** [comment]
@VFLins @czacharym Thank you for pressing the urgency, and I certainly share your view that latest major releases should be supported. Let me get the team to see what the situation is now

**Comment by czacharym** [comment]
> [@VFLins](https://github.com/VFLins) [@czacharym](https://github.com/czacharym) Thank you for pressing the urgency, and I certainly share your view that latest major releases should be supported. Let me get the team to see what the situation is now

Well, I'm sure it will be a long time before this is addressed.  Alternatives like Gradio already support 3.13 and 3.14.  Taipy is lagging behind.  Therefore, it seems its not a priority.  This will lead others to just seek out Gradio.

**Comment by VFLins** [comment]
As far as my understanding of this thread goes, Taipy is holding on to an old Pandas version to maintain compatibility with Dataiku, and this issue is blocked only for this reason. This makes supporting Dataiku a burden to Taipy, and either Daitaiku does not intend to upgrade their dependencies, or this issue is blocked for no reason.

In any case, in my opinion, the "burden" should be relieved, and whoever wants to use Taipy with Dataiku should freeze Taipy to the current version in their `pyproject.toml`.
